### ISSUE TITLE
Pytest error at set_timing_features

### ISSUE THREAD

**Issue by SeiyaNozaki** [cta-observatory/cta-lstchain#554]
I got the following `LinAlgError` error during pytest.  It seems this error was raised when timing parameters are computed if only one pixel is alive after image cleaning. For now only `ValueError` from `from ctapipe.image.timing_parameters` is defined as an exception, but not for `LinAlgError`.
I'm not sure this is the case only for me... If not, I will create PR to add LinAlgError as an exception.

```
(lst-dev) [seiya.nozaki@cp02 cta-lstchain]$ pytest lstchain/tests/test_lstchain.py
============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.8, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /fefs/aswg/workspace/seiya.nozaki/ctasoft/cta-lstchain
plugins: ordering-0.6
collected 16 items

lstchain/tests/test_lstchain.py FFF.....FF......                                                                                                                                                           [100%]

==================================================================================================== FAILURES ====================================================================================================
_________________________________________________________________________________________________ test_r0_to_dl1 _________________________________________________________________________________________________

    @pytest.mark.run(order=1)
    def test_r0_to_dl1():
        from lstchain.reco.r0_to_dl1 import r0_to_dl1
        infile = mc_gamma_testfile
>       r0_to_dl1(infile, custom_config=custom_config, output_filename=dl1_file)

lstchain/tests/test_lstchain.py:100:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
lstchain/reco/r0_to_dl1.py:448: in r0_to_dl1
    use_main_island=True)
lstchain/reco/r0_to_dl1.py:157: in get_dl1
    hillas)
lstchain/io/lstcontainers.py:183: in set_timing_features
    timepars = timing_parameters(geom, image, peak_time, hillas)

.
.
.

    def _raise_linalgerror_lstsq(err, flag):
>       raise LinAlgError("SVD did not converge in Linear Least Squares")
E       numpy.linalg.LinAlgError: SVD did not converge in Linear Least Squares

../../Install/anaconda3/envs/lst-dev/lib/python3.7/site-packages/numpy/linalg/linalg.py:100: LinAlgError
```

**Comment by moralejo** [comment]
Hi, just to say I have been getting the same for a couple of weeks. No idea what it is, I thought until now that it was a problem of my installation only.

**Comment by maxnoe** [comment]
In ctapipe we require at least five pixels in the stage 1 tool after cleaning,  so this is why we never saw that error. 

What confuses me,  is that the result seems only to happen on some machines (Mac os?).

Tests should be deterministic. Is there randomness involved somewhere in the failing test?

**Comment by SeiyaNozaki** [comment]
At least for me, I got this error on the lapalma server...

In lstchain, timing parameters are computed here if `n_pixels>0`
https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/reco/r0_to_dl1.py#L154-L157

and `keep_isolated_pixels` is `True` and `min_number_picture_neighbors` is 1 in `test_lstchain.py`. 
https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/tests/test_lstchain.py#L35-L37

So I think it could be possible that the events with `n_pixels=1` survive. `keep_isolated_pixels` should be `false`?
Actually it is `false` in `lstchain_standard_config.json`

**Comment by SeiyaNozaki** [comment]
Hi again, 
it seems a configuration of `LocalPeakWindowSum` (`window_shift` and `window_width`) is not defined in `test_lstchain.py`.
I checked it with different commits, then it is (3, 7) and sometimes (5, 12) during pytest.
(I got this error from 593aad042f6e9213850135ca725b8fe7b0f1875d which changed a configuration of `LocalPeakWindowSum` in `lstchain_standard_config.json` for example)

So I think the problem is:
1) a configuration of the integration method is not defined in `test_lstchain.py`, and it caused randomness of charge distribution with the same data set.
2) Even if `n_pixels` is 1, events are processed to compute the timing parameters, and `LinAlgError` is raised.

Do you agree?

**Comment by maxnoe** [comment]
> , and it caused randomness of charge distribution with the same data set.

Why does it cause randomness?

**Comment by maxnoe** [comment]
Ah, and we should definitely run the tests using the default config! This will also test that the default config is valid (to some point).

**Comment by SeiyaNozaki** [comment]
>Why does it cause randomness?
Well, honestly speaking I'm not sure, but it could be a possible cause of randomness. I don't know how the configuration of the integration method is set with a custom config file missing the description of the integration method.

> Ah, and we should definitely run the tests using the default config! This will also test that the default config is valid (to some point).
Yes, I agree. I will create PR for that.

**Comment by maxnoe** [comment]
>  file missing the description of the integration method.

It should use the default method, which has no element of (pseudo-)randomness as far as I'm aware. 

So it really buggles me my wthis only fails sometimes.

**Comment by moralejo** [comment]
> 
> So I think it could be possible that the events with `n_pixels=1` survive. `keep_isolated_pixels` should be `false`?
> Actually it is `false` in `lstchain_standard_config.json`

Yes, this is means a single pixel, even if above the "picture" threshold of the cleaning, will be removed if it has no neighbor which is also above that threshold. This is intended to get rid of afterpulses.

**Comment by SeiyaNozaki** [comment]
closed via #557
